docs/proposal/aws-resource-handling.md

```diff
@@ -0,0 +1,203 @@
+# AWS Resource Handling
+
+## Problem
+
+Since the AWS APIs do not provide consistent support for tagging resources on creation or idempotent operations, we need to ensure that we can provide as much of an external guarantee to users as we can. Otherwise we risk creating and orphaning resources on users leading to extraneous costs and the potential to exhaust resource quotas prematurely.
+
+## Existing solutions and drawbacks for our use
+
+### kops
+
+kops relies heavily on tagging of resources. This provides the benefit of recreating the state of the cluster resources by querying with filters. However, in many cases tagging requires a second API call and there are some resources that are not able to be tagged. If we succeed in creating a resource but fail to tag the resource, then we risk the chance of orphaning that resource for the user.
+
+### Kubicorn
+
+In contrast to the kops approach, Kubicorn mainly relies on recording the resource IDs as part of state. However, since we rely on an external API server for recording the state there is still a possibility of creating the resource and failing to record the resource ID, which still exposes the possible risk of orphaning that resource for the user.
+
+## Summary of edge cases for creating an individual resource
+
+1. resource create succeeds, but subsequent tagging fails
+2. resource creates succeeds, but update of cluster/machine object fails
+3. attempting to delete resource fails after an attempt to rollback due to a failure to record the ID of the created resource to the cluster/machine object for resources that do not support tagging on create.
+4. the controller/actuator dies after creating a resource but before tagging and or recording the resource
+
+## Misc TODOs
+
+- Solicit feedback on whether aws-sdk-go and client-go retry defaults are sufficient:
+  - aws-sdk-go
+    - https://docs.aws.amazon.com/general/latest/gr/api-retries.html
+    - https://docs.aws.amazon.com/sdk-for-go/api/aws/client/#DefaultRetryer
+  - client-go
+    - https://github.com/kubernetes/client-go/blob/master/rest/request.go#L659
```

Comment:
@kris-nova can you provide any context to why Kops leverages additional retries outside of the ones built into the aws client?